core/pkg/server/handler.go

```diff
@@ -939,6 +959,219 @@ func (h *Handler) handleTBrecord(record *service.Record) {
 	}
 }
 
+// handleHistory handles a history record. This is the main entry point for history records.
+// It is responsible for handling the history record internally, processing it,
+// and forwarding it to the Writer.
+func (h *Handler) handleHistory(history *service.HistoryRecord) {
+	// TODO replace history encoding with a map, this will make it easier to handle history
+	h.activeHistory = NewActiveHistory(
+		WithStep(history.GetStep().GetNum()),
+	)
+	h.activeHistory.UpdateValues(history.GetItem())
+
+	h.flushHistory(history)
+
+	h.activeHistory.Flush()
+}
+
+// imputeStepMetric imputes a step metric if it needs to be synced, but not part of the history record.
```

Comment:
what does impute mean in this context? is it a word we use elsewhere when talking about step metrics?

edit: i see this was moved from history.go, but same question

Reply:
oh, it is used in the context of define_metric, the step_sync argument, basically add a value for this metric even if it wasn't explicitly logged for this step.